Mathlib/CategoryTheory/MorphismProperty/Composition.lean

```diff
@@ -193,30 +193,39 @@ instance inf {P Q : MorphismProperty C} [P.IsMultiplicative] [Q.IsMultiplicative
 
 end IsMultiplicative
 
+/-- A class of morphisms `W` has the of-postcomp property if whenever
+`g` and `f ≫ g` are in `W`, also `f` is in `W`. -/
+class HasOfPostcompProperty (W : MorphismProperty C) : Prop where
+  of_postcomp {X Y Z : C} (f : X ⟶ Y) (g : Y ⟶ Z) : W g → W (f ≫ g) → W f
```

Comment:
I think it would be more useful if this is `W' g → W (f ≫ g) → W f` for some potentially different `W'`.

Reply:
I agree, I was lazy. Changed now and turns out it was not so bad. I kept `HasTwoOutOfThreeProperty` as is for now. If a more general assumption should be necessary one can just combine the correct combination of `Respects{Left, Right}` and `HasOf{Post, Pre}compProperty` needed.